docs/index.rst

```diff
@@ -35,13 +35,13 @@ eProsima Fast DDS Documentation
    /fastdds/logging/logging
    /fastdds/xml_configuration/xml_configuration
    /fastdds/env_vars/env_vars
+   /fastdds/dynamic_types/dynamic_types
```

Comment:
I'd actually put it even before Environment Variables, but we will work the order later anyway

Reply:
I'll leave it for later, then